server/src/pages/shotindex/view.js

```diff
@@ -126,23 +140,32 @@ class Body extends React.Component {
     }
 
     return (
-      <a href={shot.viewUrl} className={`shot ${this.getClipType(clip._image.dimensions)}`} key={shot.id} onClick={this.onOpen.bind(this, shot.viewUrl)}>
-        <div className="shot-image-container" style={{
-          backgroundImage: `url(${imageUrl})`
-        }}>
-        </div>
-        <div className="shot-info">
-        <div className="title-container">
-          <h4>{this.displayTitle(shot.title)}</h4>
-        </div>
-        <div className="link-container">
-          {favicon}
-          <div className="shot-url">
-            {shot.urlDisplay}
+      <div className={`shot ${this.getClipType(clip._image.dimensions)}`} key={shot.id}>
+        <a href={shot.viewUrl} onClick={this.onOpen.bind(this, shot.viewUrl)}>
+          <div className="shot-image-container" style={{
+            backgroundImage: `url(${imageUrl})`
+          }}>
           </div>
+          <div className="shot-info">
+          <div className="title-container">
+            <h4>{this.displayTitle(shot.title)}</h4>
+          </div>
+          <div className="link-container">
+            {favicon}
+            <div className="shot-url">
+              {shot.urlDisplay}
+            </div>
+          </div>
+          </div>
+        </a>
+        <div className="alt-actions-container">
+          { /* TODO: get a working download link on the next line */ }
```

Comment:
Added a download link, but couldn't make it work! we'll need to file a follow up bug, i doubt this is very hard.